python/cudf/cudf/core/frame.py

```diff
@@ -2693,7 +2693,15 @@ def acos(self):
                     1.5707963267948966,  1.266103672779499],
                     dtype='float64')
         """
-        return self._unaryop("acos")
+        if isinstance(self, cudf.Series):
+            data_type = cudf.utils.dtypes.get_min_float_dtype(self._column)
+        if isinstance(self, cudf.DataFrame):
+            for col in self._columns:
+                data_type = cudf.utils.dtypes.get_min_float_dtype(col)
+        data = self.astype(data_type)
+        data = data._unaryop("acos")
+        newdata = data.mask((data < 0) | (data > np.pi + 1))
+        return newdata
```

Comment:
```suggestion
        result = self.copy()
        for col in result._data:
            min_float_dtype = cudf.utils.dtypes.get_min_float_dtype(result._data[col])
            result._data[col] = result._data[col].astype(min_float_dtype)
        result = result._unaryop("acos")
        result = result.mask((result < 0) | (result > np.pi + 1))
        return result
```

Reply:
Incase a `cudf.DataFrame` is calling `acos` then the result will be type-casting all the columns with `data_type` which is incorrect as in this case `data_type` contains the minimum float dtype of last column only. Instead we could do this in a generic way for both series and dataframe.

Reply:
Sounds good, I'll add the commit and test it out.